www/data-types.html

```diff
@@ -0,0 +1,319 @@
+        minLevel: 9,
+        getTileUrl: function(level, x, y) {
+            return `${level}/${x}-${y}`;
+        },
+        downloadTileStart: function(imageJob) {
+            imageJob.finish("すごい！", null, "myAwesomeType");
+        },
+        downloadTileAbort: function(context) {
+            //pass
+        },
+    }
+});
+    </pre>
+    <script>
+        function getRandomColor() {
+            const letters = '0123456789ABCDEF';
+            let color = '#';
+            for (let i = 0; i < 6; i++) {
+                color += letters[Math.floor(Math.random() * 16)];
+            }
+            return color;
+        }
+
+        OpenSeadragon.convertor.learn("myAwesomeType", "myAwesomeType", (tile, text) => {
+            return text;
+        }, 0, 1);
+
+        // We don't provide convertor context2d -> myAwesomeType, since we won't be needing it
+        OpenSeadragon.convertor.learn("myAwesomeType", "context2d", (tile, text) => {
+            const canvas = document.createElement('canvas');
+            canvas.width = tile.sourceBounds.width;
+            canvas.height = tile.sourceBounds.height;
+            const context = canvas.getContext('2d');
+
+            const gradient = context.createLinearGradient(0, 0, canvas.width, canvas.height);
+            gradient.addColorStop(0, getRandomColor());
+            gradient.addColorStop(1, getRandomColor());
+
+            context.fillStyle = gradient;
+            context.fillRect(0, 0, canvas.width, canvas.height);
+
+            context.font = 'bold 30px Arial';
+            context.fillStyle = '#FFFFFF';
+            context.textAlign = 'center';
+            context.textBaseline = 'middle';
+            context.fillText(text, canvas.width / 2, canvas.height / 2);
+
+            return context;
+        }, 1, Math.sqrt(2)); // sometimes it is hard to estimate the cost, let just say
+        // the cost is sure greater than n, at least by factor of sqrt(2) - text render
+
+        OpenSeadragon({
+            id:            "example-custom-tilesource-advanced",
+            prefixUrl:     "/openseadragon/images/",
+            navigatorSizeRatio: 0.25,
+            wrapHorizontal:     true,
+            loadTilesWithAjax:  true, // no effect
+            tileSources:   {
+                height: 1024,
+                width:  1024,
+                tileSize: 256,
+                minLevel: 9,
+                getTileUrl: function(level, x, y) {
+                    return `${level}/${x}-${y}`;
+                },
+                downloadTileStart: function(imageJob) {
+                    imageJob.finish("すごい！", null, "myAwesomeType");
+                },
+                downloadTileAbort: function(context) {
+                    //pass
+                },
+            }
+        });
+    </script>
+</div>
+
+<div class="description">
+    <h3>Removing Async Support</h3>
+</div>
+OpenSeadragon supports <code>supportsAsync</code> option that can remove asynchronous execution.
```

Comment:
I think this needs a little more information. How do you use `supportsAsync`? Is that an option when you create the viewer? Or is it something you put in your data type definition? We should have an example here.

Reply:
It is a part of the basic viewer options. It should be in the API docs. We can extend on the docs too.

Reply:
I think it's worth being explicit here. Also saying that it's `supportsAsync: false` that you're talking about. You say "if you need to enable this", but I think you really mean "disable", as in setting `supportsAsync` disables async support, which is enabled by default. Am I understanding correctly?